### DISCUSSION TITLE
What does "unsigned attribute for time" in Debug log means for time attribute

### DISCUSSION THREAD

**Discussion by ghost** [dtcenter/METplus#1867] [Configuration] [answered]
Hi, I am checking netcdf validation data and I got the following message while using plot_data_plane   :  

plot_data_plane icap_2022101100_MME_modeaod550.nc taod1.ps 'name="total_aod_mean"; level="(0,*,*)";' -v 6
DEBUG 1: Opening data file: icap_2022101100_MME_modeaod550.nc
DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_NcCF".
DEBUG 4: NcCfFile::open() -> parsing units for the time variable "hours since 1900-01-01"
DEBUG 4: parse_cf_time_string() -> parsed NetCDF CF convention time unit string "hours since 1900-01-01"
DEBUG 4:                 as a reference time of 19000101_000000 and 3600 second(s) per time step.
DEBUG 6: has_unsigned_attribute() -> time does not have _Unsigned attribute.
DEBUG 4: get_nc_data(NcVar *, double *) add_offset = 0, scale_factor=1, cell_count=19, is_unsigned_value: 0 for time
DEBUG 4: apply_scale_factor(double) unpacked data: count=0 out of 19. FillValue(int)=-9999 data range [1.07623e+06 - 1.07634e+06] raw data: [1076232 - 1076340] Positive count: 19
DEBUG 4: NcCfFile::open() -> could not extract init time from the "forecast_reference_time" variable.
DEBUG 4: NcCfFile::open() -> could not extract init time from file name.
DEBUG 4: VarInfoFactory::new_var_info() -> created new VarInfo object of type "FileType_NcCF".
DEBUG 6: NcCfFile::getData(NcVar *, const LongArray &, DataPlane &) const -> took 0.000948 seconds
DEBUG 4: NcCfFile::getData() -> setting the unset init time to the valid time of 20221011_000000.
DEBUG 4: 
DEBUG 4: Grid information:
DEBUG 4:    Projection: Lat/Lon
DEBUG 4:    Nx: 360
DEBUG 4:    Ny: 180
DEBUG 4:    lat_ll: -89.500
DEBUG 4:    lon_ll: 179.500
DEBUG 4:    delta_lat: 1.000
DEBUG 4:    delta_lon: 1.000
DEBUG 4:    wrapLon: true
DEBUG 4: 
DEBUG 4: Data plane information:
DEBUG 4:    plane min: 0.00624574
DEBUG 4:    plane max: 2.23655
DEBUG 4:    valid time: 20221011_000000
DEBUG 4:    lead time: 000000
DEBUG 4:    init time: 20221011_000000
DEBUG 4:    accum time: 000000
DEBUG 4: 
DEBUG 1: Creating postscript file: taod1.ps
DEBUG 4: draw_map() -> Plotting map data file: /apps/ops/para/libs/intel/19.1.3.304/met/10.1.1/share/met/map/country_data
DEBUG 4: draw_map() -> Plotting map data file: /apps/ops/para/libs/intel/19.1.3.304/met/10.1.1/share/met/map/usa_state_data
DEBUG 4: draw_map() -> Plotting map data file: /apps/ops/para/libs/intel/19.1.3.304/met/10.1.1/share/met/map/major_lakes_data

Any clue. Thanks

**Comment by j-opatz** [comment, marked as answer]
Hello,

And thank you for your question. At its core, the statement,

`has_unsigned_attribute() -> time does not have _Unsigned attribute`

is just that; the `time` variable in the input file does not have an attribute called Unsigned.

A slightly more technical side of this is what Unsigned and Signed mean. Signedness are ways of declaring a range of allowable values for the variable type. Unsigned restricts it to zero and nonnegative, while Signed allows the full value range.

You'll note that these types of messages appear in the `Debug 6` level, which is loaded with technical aspects that are not immediately relevant to the correct reading and verification of data. If you run this configuration often and are not troubleshooting, I would suggest keeping the verbosity to no higher than 2.
